app/services/grade_capstone_evaluations.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+require 'salesforce_api'
+
+# Responsible for calculating the grades for the Capstone Evaluation Teamwork assignment and sending grades to Canvas
+class GradeCapstoneEvaluations
+  def initialize(course, lti_launch, new_capstone_eval_submissions, all_capstone_eval_submissions)
+    @course = course
+    @lti_launch = lti_launch
+    @new_capstone_eval_submissions = new_capstone_eval_submissions
+    @all_capstone_eval_submissions = all_capstone_eval_submissions
+  end
+
+  def run
+    users_to_grade = CapstoneEvaluationSubmissionAnswer.where(submission: @new_capstone_eval_submissions)
+      .map { |answer| answer.for_user }
+      .uniq
+
+    grades = {}
+    users_to_grade.each do |user|
+      submission_answers_for_user = CapstoneEvaluationSubmissionAnswer.where(
+        submission: @all_capstone_eval_submissions,
+        for_user: user
+      )
+
+      question_scores = CapstoneEvaluationResultsController.new.grade_capstone_eval_questions(submission_answers_for_user)
+      total_score = question_scores.map { |k, v| v }.sum
+
+      grades[user.canvas_user_id] = total_score
+    end
+
+    CanvasAPI.client.update_grades(@course.canvas_course_id, @lti_launch.assignment_id, grades)
+
+    @new_capstone_eval_submissions.each do |submission|
+      submission.update!(new: false)
+    end
```

Comment:
nit: you can also do 
```suggestion
    @new_capstone_eval_submissions.update_all(new: false)
    end
```
It will skip callbacks/validation https://apidock.com/rails/ActiveRecord/Base/update_all/class, but I think that's OK in this case since those submissions should already be guaranteed valid, since they're in the DB?

Reply:
ah okay nice, just updated to this